Ruteri/dummy-tdx-dcap
# Adds verification endpoint

## 📝 Summary

Adds a `/verify` endpoint, which accepts POST with the raw quote as body. Returns 200 OK only when the quote verification passes, and returns the quote parsed as json for convenience. Responds with 401 for malformed requests and 418 ([I'm a teapot](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/418)) if request is correctly formed but verification does not pass, and returns the reason in the body.
Renames `/attestation` to just `/attest`.

To test:
1. run the server `go run cmd/httpserver/main.go`
2. verify with `go run cmd/cli-verify/main.go --quote $(cat httpserver/sample_quote.hex)`

## ⛱ Motivation and Context

It's cool

## 📚 References

---

## ✅ I have run these commands

* [x] `go mod tidy`
